core/src/test/java/org/elasticsearch/search/highlight/HighlightBuilderTests.java

```diff
@@ -287,44 +292,73 @@ public MappedFieldType fieldMapper(String name) {
         for (int runs = 0; runs < NUMBER_OF_TESTBUILDERS; runs++) {
             HighlightBuilder highlightBuilder = randomHighlighterBuilder();
             SearchContextHighlight highlight = highlightBuilder.build(mockShardContext);
-            XContentBuilder builder = XContentFactory.contentBuilder(randomFrom(XContentType.values()));
-            if (randomBoolean()) {
-                builder.prettyPrint();
-            }
-            builder.startObject();
-            highlightBuilder.innerXContent(builder);
-            builder.endObject();
-            XContentParser parser = XContentHelper.createParser(builder.bytes());
-
-            SearchContextHighlight parsedHighlight = new HighlighterParseElement().parse(parser, mockShardContext);
-            assertNotSame(highlight, parsedHighlight);
-            assertEquals(highlight.globalForceSource(), parsedHighlight.globalForceSource());
-            assertEquals(highlight.fields().size(), parsedHighlight.fields().size());
-
-            Iterator<org.elasticsearch.search.highlight.SearchContextHighlight.Field> iterator = parsedHighlight.fields().iterator();
             for (org.elasticsearch.search.highlight.SearchContextHighlight.Field field : highlight.fields()) {
```

Comment:
I'd probably make this just `for (Field field : highlight.fields()) {` while I was here.